[NEW] Display Keyboard when launching dialog

HarveyDoom opened the pull request:
First ever pull request.
requestFocus() would not work on its own. Setting the Soft Input Mode gets keyboard to display.

<!-- INSTRUCTION: Keep the line below to notify all core developers about this new PR -->
@RocketChat/android

<!-- INSTRUCTION: Inform the issue number that this PR closes, or remove the line below -->
Closes #331 

<!-- INSTRUCTION: Tell us more about your PR with screen shots if you can -->

HarveyDoom committed:
#331 - Display Keyboard when launching dialog.

First ever pull request.
requestFocus() would not work on its own. Setting the Soft Input Mode gets keyboard to display.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/RocketChat/Rocket.Chat.Android?pullRequest=500) <br/>All committers have signed the CLA.

HarveyDoom committed:
Added requestFocus(View view)

requestFocus can be called by child classes to force keyboard to display when a view requests Focus.

HarveyDoom committed:
Moved requestFocus() to parent class.

HarveyDoom committed:
#331 - Keyboard focus to Direct Message Fragment

philipbrito reviewed (changes requested):
Hi @HarveyDoom!

In order to merge your PR you need to fix a few things. Here is a tip:

- Import the `WindowManager` class to use [here](https://github.com/HarveyDoom/Rocket.Chat.Android/blob/a1534394ddda689ff25a75441b70872b5a2678c0/app/src/main/java/chat/rocket/android/fragment/sidebar/dialog/AbstractAddRoomDialogFragment.java#L47).
- Import the `TextInputEditText` class to use [here](https://github.com/HarveyDoom/Rocket.Chat.Android/blob/a1534394ddda689ff25a75441b70872b5a2678c0/app/src/main/java/chat/rocket/android/fragment/sidebar/dialog/AddChannelDialogFragment.java#L42).

HarveyDoom committed:
Added Import Statement

HarveyDoom committed:
Added import statement

HarveyDoom committed:
Merge branch 'develop' into develop

leonardoaramaki committed:
Merge branch 'develop' into develop

leonardoaramaki reviewed (approved):
@HarveyDoom Thank you a lot for your contribution! 
Sorry for the long time gap to review this one.
Merging!
